pythonforandroid/bootstrap.py

```diff
@@ -138,20 +141,22 @@ def name(self):
         return modname.split(".", 2)[-1]
 
     def prepare_build_dir(self):
-        '''Ensure that a build dir exists for the recipe. This same single
-        dir will be used for building all different archs.'''
+        """Ensure that a build dir exists for the recipe. This same single
+        dir will be used for building all different archs."""
         self.build_dir = self.get_build_dir()
-        self.common_dir = self.get_common_dir()
-        copy_files(join(self.bootstrap_dir, 'build'), self.build_dir)
-        copy_files(join(self.common_dir, 'build'), self.build_dir,
-                   override=False)
-        if self.ctx.symlink_java_src:
-            info('Symlinking java src instead of copying')
-            shprint(sh.rm, '-r', join(self.build_dir, 'src'))
-            shprint(sh.mkdir, join(self.build_dir, 'src'))
-            for dirn in listdir(join(self.bootstrap_dir, 'build', 'src')):
-                shprint(sh.ln, '-s', join(self.bootstrap_dir, 'build', 'src', dirn),
-                        join(self.build_dir, 'src'))
+
+        # get all bootstrap names along the __mro__, cutting off Bootstrap and object
+        classes = self.__class__.__mro__[:-2]
+        bootstrap_names = [cls.name for cls in classes] + ['common']
+        bootstrap_dirs = [
+            join(self.ctx.root_dir, 'bootstraps', bootstrap_name)
+            for bootstrap_name
+            in reversed(bootstrap_names)
```

Comment:
Prefer to merge this with the previous line, it's an unnecessary line break.

Reply:
done